bin/all_sky_search/pycbc_rerank_dq

```diff
@@ -0,0 +1,124 @@
+rate_group = parser.add_mutually_exclusive_group()
+rate_group.add_argument("--rate-file", type = str, default = None,
+                        help="File containing the mapping from each "
+                             "dq bin to the trigger rate")
+rate_group.add_argument("--bank-file", type = str, default = None,
+                        help="File containing the template bank")
+parser.add_argument("--dq-type", type = str, default = 'dq',
+                     help="Name of dq type to be included in output file "
+                          "metadata [default='dq']")
+parser.add_argument("--output-file", required=True,
+                     help="name of output file")
+
+args = parser.parse_args()
+pycbc.init_logging(args.verbose)
+
+ifo = args.ifo
+
+log_like = numpy.array([])
+times = numpy.array([])
+for filename in args.input_file:
+    logging.info('Reading file %s...', filename)
+    dq_data = load_timeseries(filename, group=args.input_channel)
+    log_like = numpy.concatenate((log_like,dq_data[:]))
+    times = numpy.concatenate((times,dq_data.get_sample_times()))
+    del dq_data
+
+log_like_ranking = numpy.argsort(log_like)
+len_log_like = len(log_like)
+
+times = times[log_like_ranking]
+log_like_vals = numpy.array(log_like[log_like_ranking])
+
+log_like_dict = {}
+percent_dict = {}
+
+if args.rate_file is not None:
+    rates = {}
+    locs_dict = {}
+    if path.isfile(args.rate_file):
+        with h5py.File(args.rate_file,'r') as f:
+            bin_names = f.attrs['names'][:]
+            for bin_name in bin_names:
+                logging.info('Processing bin %s...', bin_name)
+                rates = f['%s/rates'%bin_name][:]
+                locs_dict[bin_name] = f['%s/locs'%bin_name][:]
+                log_like_dict[bin_name] = numpy.zeros(len(log_like))
+
+                if not args.no_low_dq_cutoff:
+                    rates = numpy.array([max(r,1.) for r in rates])
+
+                log_rates = numpy.log(rates)
+                percent_dict[bin_name] = log_rates
+
+                n_bins = len(log_rates)
+                percentiles = numpy.linspace(0,100,n_bins+1)
+                dq_percentiles = numpy.percentile(log_like_vals,percentiles)[1:]
+
+                rates_bin = numpy.array([n_bins - \
+                                  len(dq_percentiles[dq_percentiles >= dq_ll]) \
+                                  for dq_ll in log_like_vals])
+                log_like_dict[bin_name] = log_rates[rates_bin.astype('int')]
+
+
+    else: logging.error('Could not find binned trigger rates file')
+else:
+    # Use analytic model from Godwin et al. arXiv:2010.15282 
+    bin_names = ['all_bin']
+    with h5py.File(args.bank_file, 'r') as bank:
+        locs_dict = {'all_bin': numpy.arange(0, len(bank['mass1'][:]), 1)}
+
+    log_like = numpy.arange(len(log_like))*100./len(log_like)
+
+    p_min = int(len_log_like/2.)
+    p_max = int(len_log_like/(1.+numpy.e**-args.max_stat))
```

Comment:
Unless there is a numerical reason to prefer the explicit power, I suggest using `numpy.exp()` here.